src/vsphere_cpi/spec/unit/cloud/vsphere/vcenter_client_spec.rb

```diff
@@ -177,9 +194,25 @@ module VSphereCloud
         end
       end
 
+
+      context 'when file manager raises "Invalid datastore path" error for the disk' do
+        it 'does not raise error' do
+          expect(task_runner).to receive(:run) do |*args, &block|
+            expect(block.call).to eq(vmdk_task)
+          end.and_raise(RuntimeError.new('Invalid datastore path some/path'))
+
+          expect(virtual_disk_manager).to receive(:delete_virtual_disk).
+              with('[some-datastore] some/path', datacenter).
+              and_return(vmdk_task)
+          expect {
```

Comment:
No need to `expect` ... `to_not` ... `raise_error`. Just remove the entire expectation and replace it with a comment. If an error is raised the test will fail.